amr-wind/equation_systems/icns/source_terms/BodyForce.cpp

```diff
@@ -16,18 +17,34 @@ namespace amr_wind::pde::icns {
 BodyForce::BodyForce(const CFDSim& sim) : m_time(sim.time()), m_mesh(sim.mesh())
 {
 
-    // Read the geostrophic wind speed vector (in m/s)
+    // Body Force arguments
     amrex::ParmParse pp("BodyForce");
     pp.query("type", m_type);
     m_type = amrex::toLower(m_type);
+    bool no_type_specified = !pp.contains("type");
+    bool file_specified = pp.contains("uniform_timetable_file");
 
+    // Prepare type of body force distribution
     if (m_type == "height-varying") {
+        // Constant in time, varies with z
         pp.get("bodyforce-file", m_bforce_file);
         read_bforce_profile(m_bforce_file);
+    } else if (
+        m_type == "uniform_timetable" ||
+        (no_type_specified && file_specified)) {
+        // Still used if type not specified but file is
+        // Uniform in space, varies with time
+        pp.get("uniform_timetable_file", m_utt_file);
+        read_bforce_timetable(m_utt_file);
     } else {
         pp.getarr("magnitude", m_body_force);
         if (m_type == "oscillatory") {
             pp.get("angular_frequency", m_omega);
+        } else if (m_type != "uniform") {
+            amrex::Abort(
+                "BodyForce type not supported. Please choose uniform "
+                "(default), height-varying, oscillatory, or "
+                "uniform_timetable.\n");
```

Comment:
Maybe this is just me, but it seems somewhat confusing to differentiate between `uniform` and `uniform_timetable`. The latter is supposed to be uniform in `z` but time varying?

Reply:
I'm using "uniform" as the default, but maybe that should be renamed "uniform_constant". The default is that the user specifies a single force vector that is applied over the entire domain for the entire simulation. The new capability is to have that uniform force vary in time ("uniform_timetable"). Other options that were already in the code were oscillating (where the force is applied uniformly, but it varies in time according to a sinusoid) and height-profile (constant in time but varies in z).

Reply:
I'm okay with either option as long as it's easy to know what each corresponds to in the documentation. Personally I have a slight preference for `uniform_constant` since it makes it a bit clearer.

Reply:
I like your "ultra minor" comment, but I'm pretty sure that underscores are used much more commonly in AMR-Wind parsing arguments. So, I'm putting in changes to make it consistent with underscores but remain backwards-compatible with hyphens for the height-varying arguments. Maybe that will be sufficient?